### PR TITLE
Simplify DI for base classes & add error service back

### LINKED ISSUE #1081: Action errors are no longer displayed to the user
Regression from #989: Previously, if an action request returned with an error code, the error message was displayed to the user via a snack bar. Now, if I try to change my password for example and enter the wrong old password, the request fails silently. I suspect that this affects all requests and not only the password change.

### LINKED ISSUE #1225: Enhance the DI
Many components and services are extended from a base directive. The base directive usally injects services, which have to be injected in the extended component or service. The DI can be enhanced as shown in #1081.

### PR DESCRIPTION
**Pull request by jsangmeister**
fixes #1225

### CODE REVIEW

File: client/src/app/site/base/base-controller.ts [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
+    public translate: TranslateService;
 
-    public constructor(
-        protected controllerServiceCollector: ControllerServiceCollectorService,
-        protected baseModelCtor: ModelConstructor<M>,
-        protected repo: BaseRepository<V, M>
-    ) {}
+    public constructor(protected baseModelCtor: ModelConstructor<M>, protected repo: BaseRepository<V, M>) {
+        const injector = AppInjector.getInjector();
+        this.translate = injector.get(TranslateService);
```
Comment (GabrielInTheWorld) [inline client/src/app/site/base/base-controller.ts:16]: Here we have to check, if this is sufficient to get the translations extracted from the TypeScript files. To check this, run `npm run extract` and check, if there are some changes. This might be related to #1112.

File: client/src/app/site/base/base-import.service/base-import.service.ts [hunk 2]
```diff
@@ -217,13 +214,19 @@ export abstract class BaseImportService<MainModel extends Identifiable> implemen
...
 
     /**
      * Constructor. Creates a fileReader to subscribe to it for incoming parsed
      * strings
      */
-    public constructor(private importServiceCollector: ImportServiceCollectorService) {
+    public constructor() {
+        const injector = AppInjector.getInjector();
+        this._papa = injector.get(Papa);
+        this.translate = injector.get(TranslateService);
```
Comment (GabrielInTheWorld) [inline client/src/app/site/base/base-import.service/base-import.service.ts:228]: This is the same as for `BaseControllerService`.

File: client/src/app/site/pages/meetings/modules/poll/components/poll-form/poll-form.component.ts [hunk 3]
```diff
@@ -146,7 +146,7 @@ export class PollFormComponent extends BaseUiComponent implements OnInit {
             //         this.data.max_votes_amount = assignment.open_posts;
             //     }
             //     if (!this.data.pollmethod) {
-            //         this.data.pollmethod = this.meetingSettingService.instant(`assignment_poll_default_method`)!;
+            //         this.data.pollmethod = this.meetingSettingsService.instant(`assignment_poll_default_method`)!;
             //     }
             // }
```
Comment (GabrielInTheWorld) [inline client/src/app/site/pages/meetings/modules/poll/components/poll-form/poll-form.component.ts:151]: I know, its something from #1085, but I think, you can remove this ;-)

File: client/tsconfig.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
         "baseUrl": "./",
         "declaration": false,
         "downlevelIteration": true,
-        "emitDecoratorMetadata": true,
+        "emitDecoratorMetadata": false,
```
Comment (GabrielInTheWorld) [inline client/tsconfig.json:8]: Why did you change this?

File: client/src/app/openslides-main-module/services/app-injector.service.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+export class AppInjector {
+    private static injector: Injector;
+
+    static setInjector(injector: Injector) {
+        AppInjector.injector = injector;
+    }
+
+    static getInjector(): Injector {
+        return AppInjector.injector;
+    }
```
Comment (GabrielInTheWorld) [inline client/src/app/openslides-main-module/services/app-injector.service.ts:12]: In every file, you use the `Injector`, you write something like
```js
    const injector = AppInjector.getInjector();
    const service = injector.get(ServiceToken);
```

Could you provide some methods directly, which invokes the methods provided by the `Injector`?
Something like:

```suggestion
    public static get<Token, Service>(token: Token): Service {
        return AppInjector.injector.get(token);
    }
```

File: client/src/app/openslides-main-module/services/app-injector.service.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { Injector } from '@angular/core';
+
+export class AppInjector {
+    private static injector: Injector;
+
+    static setInjector(injector: Injector) {
```
Comment (GabrielInTheWorld) [inline client/src/app/openslides-main-module/services/app-injector.service.ts:6]: Please do not forget the visibility ;-)

File: client/src/app/openslides-main-module/services/app-load.service.ts [hunk 7]
```diff
@@ -54,12 +55,19 @@ const appConfigs: AppConfig[] = [
...
+
     public constructor(
         private injector: Injector,
-        private modelMapper: CollectionMapperService,
-        private mainMenuService: MainMenuService,
-        private fallbackRoutesService: FallbackRoutesService
-    ) {}
+    ) {
+        // Circumventing the standard CI here to set the injector before any other services are loaded
+        AppInjector.setInjector(injector);
```
Comment (GabrielInTheWorld) [inline client/src/app/openslides-main-module/services/app-load.service.ts:66]: Hmm, this seems a little bit hacky, doesn't it? Why don't you set the injector as in your [example](https://devblogs.microsoft.com/premier-developer/angular-how-to-simplify-components-with-typescript-inheritance/)?

```suggestion
// in main.ts
platformBrowserDynamic()
    .bootstrapModule(OpenSlidesMainModule)
    .then(moduleRef => AppInjector.setInjector(moduleRef.injector))
    .catch(err => console.error(err));
```

**Review by GabrielInTheWorld — CHANGES_REQUESTED** [review]
Overall, it looks very cool! There are some questions or remarks to resolve.

File: client/src/app/openslides-main-module/services/app-load.service.ts (hunk 7 above)
Comment (jsangmeister) [reply client/src/app/openslides-main-module/services/app-load.service.ts:66]: Because this way, angular loads some components before the bootstrap promise resolves, which fails because the injector is not present yet. This was the best place I found to set the injector before anything else is loaded.

File: client/tsconfig.json (hunk 4 above)
Comment (jsangmeister) [reply client/tsconfig.json]: I came across the same error as in #1174 and read that this might fix it, since setting this option to `true` is no longer needed since angular 8. But it didn't help, so I'll revert it for now.

File: client/src/app/openslides-main-module/services/app-injector.service.ts (hunk 5 above)
Comment (jsangmeister) [reply client/src/app/openslides-main-module/services/app-injector.service.ts]: Good call, changed it accordingly.

File: client/src/app/site/base/base-controller.ts (hunk 1 above)
Comment (jsangmeister) [reply client/src/app/site/base/base-controller.ts:17]: You are right, this destroys the extractability of strings. We have multiple options now as I see it:
- use another extraction tool (probably not practical)
- fork https://github.com/biesbjerg/ngx-translate-extract and implement a fix for this (maybe some more work, but probably the cleanest solution - unfortunately, the repo seems to be no longer maintained, which means on the other hand that we do not need to update the fork anytime soon)
- explicitly override the `translate` variable in all subclasses/use the traditional DI for the injection of the `TranslateService` (also some work, but very redundant one, but the solution is not as clean as the fork one)
